### PR TITLE
give cloudwatch event permission to launch task

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Scheduled event failing to start the performance test :/

Looks like it needs another policy?
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/CWE_IAM_role.html

# Test instructions | Instructions pour tester la modification

# Help requested | Aide requise

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: aws/performance-test/cloudwatch.tf [hunk 1]
```diff
@@ -20,8 +20,13 @@ resource "aws_cloudwatch_event_target" "ecs_scheduled_task" {
     task_definition_arn = aws_ecs_task_definition.perf_test_task.arn
 
     network_configuration {
-      subnets         = var.vpc_public_subnets
-      security_groups = [aws_security_group.perf_test.id]
+      subnets          = var.vpc_public_subnets
+      security_groups  = [aws_security_group.perf_test.id]
+      assign_public_ip = true
```
Comment (jimleroyer) [inline aws/performance-test/cloudwatch.tf:25]: We definitely need this option to be `true` and by default it was `false`.

File: aws/performance-test/task_role.tf [hunk 2]
```diff
@@ -18,7 +18,7 @@ resource "aws_iam_role_policy_attachment" "perf_test_s3_attach" {
 
 data "aws_iam_policy_document" "ecs_task_assume" {
   statement {
-    actions = ["sts:AssumeRole"]
+    actions = ["sts:AssumeRole", "iam:PassRole"]
```
Comment (jimleroyer) [inline aws/performance-test/task_role.tf:21]: I don't think that might work. The role needs to be defined on the target role defined on the event bridge and this is for the task role on the ECS task, i.e. ECS task != event bridge rule target.

When I check the configuration on the event bridge rule target, I'd need to see something here but it's blank:
image

The created JSON policy for the event bridge rule target that works should be similar to this:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ecs:RunTask"
            ],
            "Resource": [
                "arn:aws:ecs:*:239043911459:task-definition/perf-tests"
            ],
            "Condition": {
                "ArnLike": {
                    "ecs:cluster": "arn:aws:ecs:*:239043911459:cluster/perf-tests"
                }
            }
        },
        {
            "Effect": "Allow",
            "Action": "iam:PassRole",
            "Resource": [
                "*"
            ],
            "Condition": {
                "StringLike": {
                    "iam:PassedToService": "ecs-tasks.amazonaws.com"
                }
            }
        }
    ]
}
```

File: env/staging/performance-test/terragrunt.hcl [hunk 3]
```diff
@@ -39,7 +39,7 @@ inputs = {
   vpc_id                                      = dependency.common.outputs.vpc_id
   aws_pinpoint_region                         = "ca-central-1"
 
-  billing_tag_key                             = "CostCentre"
+  billing_tag_key                             = "CostCenter"
```
Comment (jimleroyer) [inline env/staging/performance-test/terragrunt.hcl:42]: I changed it back to `CostCenter` as it is this way for all of Notify's created resource. Let's change it to the French Canadian spelling in one sweep later on, as long as it is consistent throughout the code base.

File: aws/performance-test/cloudwatch.tf [hunk 4]
```diff
@@ -17,11 +17,16 @@ resource "aws_cloudwatch_event_target" "ecs_scheduled_task" {
     launch_type         = "FARGATE"
     platform_version    = "1.4.0"
     task_count          = 1
-    task_definition_arn = aws_ecs_task_definition.perf_test_task.arn
+    task_definition_arn = aws_iam_role.scheduled_task_perf_test_event_role.arn
```
Comment (jimleroyer) [inline aws/performance-test/cloudwatch.tf:20]: I changed the role for a new one specific to event bridge rule target. I followed an example [I found from Calvin over here](https://github.com/cds-snc/covid-alert-metrics-terraform/blob/657e36305e29ec249c2474eb38663cc1176edc72/aws/ecs/cloudwatch_event_role.tf).

File: aws/performance-test/cloudwatch_event_role.tf [hunk 5]
```diff
@@ -0,0 +1,37 @@
+data "aws_iam_policy_document" "scheduled_task_perf_test_event_role_assume_role_policy" {
```
Comment (jimleroyer) [inline aws/performance-test/cloudwatch_event_role.tf:1]: All of this config comes from an example [I found from Calvin over here](https://github.com/cds-snc/covid-alert-metrics-terraform/blob/657e36305e29ec249c2474eb38663cc1176edc72/aws/ecs/cloudwatch_event_role.tf). The configuration seemed to replicate what I was able to manually create through clicksops and for which AWS graciously and automatically configured for me (not in TF though!).

**Review by jimleroyer — COMMENTED** [review]
I think we need to remove the `iam:PassRole` from the `ecs_task_assume` policy document before merging, as the new defined role would add it in a dedicated role/policy now.

**Review by jimleroyer — APPROVED** [review]
LGMT 🤞

File: aws/performance-test/cloudwatch.tf (hunk 4 above)
Comment (sastels) [reply aws/performance-test/cloudwatch.tf:21]: hrm, this caused apply to crash - It's looking for the task arn here, not a role. I think the role should be in the `role_arn` near line 15

**Pull request merged**